cmd/internal/ecs/add.go

```diff
@@ -0,0 +1,390 @@
+	reportStep("Get ECS Cluster and Region")
+	printer.Infof("Searching all regions for ECS clusters. This may take a minute to complete.\n")
+
+	arnToRegion := make(map[arn]string, 0)
+	arnToName := make(map[arn]string, 0)
+	for _, region := range wf.awsRegions {
+		wf.createClient(region)
+		clusters, err := wf.listECSClusters()
+		if err != nil {
+			printer.Warningf("Skipping region %q, error: %v\n", region, err)
+			continue
+		}
+		if len(clusters) > 0 {
+			printer.Infof("Found %d clusters in region %q.\n", len(clusters), region)
+			for a, n := range clusters {
+				arnToRegion[a] = region
+				arnToName[a] = n
+			}
+		}
+	}
+
+	choices := make([]string, 0, len(arnToName))
+	for c, _ := range arnToName {
+		choices = append(choices, string(c))
+	}
+	sort.Slice(choices, func(i, j int) bool {
+		return choices[i] < choices[j]
+	})
+
+	var clusterAnswer string
+	err = survey.AskOne(
+		&survey.Select{
+			Message: "In which cluster does your application run?",
+			Help:    "Select ECS cluster with the task you want to modify.",
+			Options: choices,
+			Description: func(value string, _ int) string {
+				name := arnToName[arn(value)]
+				if name == "" {
+					return ""
+				}
+				return name + " in " + arnToRegion[arn(value)]
+			},
+		},
+		&clusterAnswer,
+	)
+
+	wf.ecsClusterARN = arn(clusterAnswer)
+	wf.createClient(arnToRegion[wf.ecsClusterARN])
+	wf.awsRegion = arnToRegion[wf.ecsClusterARN]
+	wf.ecsCluster = arnToName[wf.ecsClusterARN]
+
+	return awf_done()
+}
+
+// Find all ECS clusters in the selected region.
+func getClusterState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowState], err error) {
+	reportStep("Get ECS Cluster")
+
+	if ecsClusterFlag != "" {
+		// TODO: lookup arn
+		return awf_error(fmt.Errorf("Cluster flag is unimplemented"))
+	}
+
+	clusters, listErr := wf.listECSClusters()
+	if listErr != nil {
+		var uoe UnauthorizedOperationError
+		if errors.As(listErr, &uoe) {
+			// Permissions error, pick a different profile or region (or quit.)
+			printer.Errorf("The provided credentials do not have permission to perform %s on ECS in region %s.\n",
+				uoe.OperationName, wf.awsConfig.Region)
+			printer.Infof("Please pick a different profile or region, or assign this permission in AWS IAM.\n")
+			return awf_next(getProfileState)
+		}
+		printer.Errorf("Could not list ECS clusters: %v\n", listErr)
+		return awf_error(errors.New("Error while listing ECS clusters; try using the --cluster flag instead."))
+	}
+
+	if len(clusters) == 0 {
+		printer.Errorf("Could not find any ECS clusters in this region. Please select a different one or hit Ctrl+C to exit.\n")
+		return awf_next(getRegionState)
+	}
+
+	printer.Infof("Found %d clusters in region %q.\n", len(clusters), wf.awsRegion)
+
+	choices := make([]string, 0, len(clusters))
+	for c, _ := range clusters {
+		choices = append(choices, string(c))
+	}
+	choices = append(choices, goBackOption)
```

Comment:
```suggestion
	choices := make([]string, 0, len(clusters))
	for c, _ := range clusters {
		choices = append(choices, string(c))
	}
	choices = append(choices, goBackOption)
	sort.Strings(choices)
```

Reply:
I'll make this choice but the goBackOption should not be sorted.

Reply:
Fixed in https://github.com/akitasoftware/akita-cli/pull/175/commits/15c83ddcd47ed5309971ccd116768b1b1c2776bb